### DISCUSSION TITLE
Workbook open password

### DISCUSSION THREAD

**Discussion by SergeDubovsky** [dfinke/ImportExcel#1526] [Q&A]
Hi team! 
I can't figure out a way to set this password:
![image](https://github.com/dfinke/ImportExcel/assets/1774209/578ca952-56bd-44fa-b901-b043a6eb7e1b)
The switch in the Export-Excel -password sets the protection on sheet data, but not on the whole file. 
Is there a way to set it?

Thank you!

**Comment by SergeDubovsky** [comment]
I tried the EPP thing: https://github.com/EPPlusSoftware/EPPlus/wiki/Encrypt-and-Protect-your-workbook
```
$Excel.Workbook.Protection.SetPassword("Test")
$Excel.Encryption.IsEncrypted = $true


```
It fails on Save: Exception calling "Save" with "0" argument(s): "Error saving file

**Comment by SergeDubovsky** [comment]
It's the Powershell version. Under PS7+ it fails to load something from Crypto. It works fine under PS5

**Reply by dfinke** [reply]
Please search the issues. I think this occurred. Don't remember if there is a resolution.

**Reply by MarkE0** [reply]
Was the below message similar to your issue? I'm just getting this now in PowerShell, while it's working in Windows PowerShell:

`Could not load type 'System.Security.Cryptography.RIPEMD160Managed'`

Had tried to open a protected workbook and supplied the Password parameter.
